libopendroneid/wifi.c

```diff
@@ -17,21 +17,43 @@ sw@simonwunderlich.de
 #include <stdarg.h>
 #include <errno.h>
 #include <byteswap.h>
+#include <time.h>
 
 #include "opendroneid.h"
 
 #if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
 #define cpu_to_le16(x)  (x)
+#define cpu_to_le64(x)  (x)
 #else
 #define cpu_to_le16(x)      (bswap_16(x))
+#define cpu_to_le64(x)      (bswap_64(x))
 #endif
 
 #define IEEE80211_FCTL_FTYPE          0x000c
 #define IEEE80211_FCTL_STYPE          0x00f0
 
 #define IEEE80211_FTYPE_MGMT            0x0000
 #define IEEE80211_STYPE_ACTION          0x00D0
+#define IEEE80211_STYPE_BEACON          0x0080
+
+/* Neighbor Awareness Networking Specification v3.1 in section 2.8.2
+ * The NAN Cluster ID is a MAC address that takes a value from
+ * 50-6F-9A-01-00-00 to 50-6F-9A-01-FF-FF and is carried in the A3 field of
+ * some of the NAN frames. The NAN Cluster ID is randomly chosen by the device
+ * that initiates the NAN Cluster.
+ */
+uint8_t cluster_id[6] = { 0x50, 0x6F, 0x9A, 0x01, 0x00, 0x00 };
+int cluster_id_generated = 0;
```

Comment:
It would be preferable to not have global variables.
This could be moved inside "generate_nan_cluster_id" and be made static. Then it doesn't have to be a global variable.

Reply:
Ok 👍